opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/requisition/RequisitionNode.java

```diff
@@ -60,6 +59,9 @@
 @XmlRootElement(name = "node")
 public class RequisitionNode {
 
+    // matches any string containing :?/\'"&
+    private static final String FOREIGN_ID_REGEX = ".*[:&?/\\\\\"'\"\\*].*";
```

Comment:
Shouldn't we turn this around and test for the set of valid characters? (aka is 💩 a valid foreign ID)

Reply:
I agree.